File: 0590-n-ary-tree-postorder-traversal/0590-n-ary-tree-postorder-traversal.cpp
/*
// Definition for a Node.
class Node {
public:
    int val;
    vector<Node*> children;

    Node() {}

    Node(int _val) {
        val = _val;
    }

    Node(int _val, vector<Node*> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
public:
    vector<int>ans;

    void solve(Node* root){

        while(root->children.size()>0){
            solve(root->children[0]);
           root->children.erase(root->children.begin());
        }
        ans.push_back(root->val);

    }
    vector<int> postorder(Node* root) {
        
        if(!root) return ans;
        solve(root);

        return ans;
    }
};

File: 0590-n-ary-tree-postorder-traversal/0590-n-ary-tree-postorder-traversal.java
/*
// Definition for a Node.
class Node {
    public int val;
    public List<Node> children;

    public Node() {}

    public Node(int _val) {
        val = _val;
    }

    public Node(int _val, List<Node> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
    public 
        List<Integer>ans=new ArrayList<>();

        void solve(Node root) {
        while (root.children.size() > 0) {
            solve(root.children.get(0));  
            root.children.remove(0);      
        }
        ans.add(root.val); 
        }
    List<Integer> postorder(Node root) {
        
        if((root==null)) return ans;
        solve(root);

        return ans;
    }
}